src/main/java/org/junit/runner/manipulation/Sorter.java

```diff
@@ -33,16 +38,50 @@ public Sorter(Comparator<Description> comparator) {
     }
 
     /**
-     * Sorts the test in <code>runner</code> using <code>comparator</code>
+     * Sorts the tests in <code>target</code> using <code>comparator</code>.
      */
-    public void apply(Object object) {
-        if (object instanceof Sortable) {
-            Sortable sortable = (Sortable) object;
+    @Override
+    public void apply(Object target) {
+        /*
+         * Note that all runners that are Orderable are also Sortable (because
+         * Orderable extends Sortable). Sorting is more efficient than ordering,
+         * so we override the parent behavior so we sort instead.
+         */
+        if (target instanceof Sortable) {
+            Sortable sortable = (Sortable) target;
             sortable.sort(this);
         }
     }
 
     public int compare(Description o1, Description o2) {
         return comparator.compare(o1, o2);
     }
+ 
+    /**
+     * {@inheritDoc}
+     *
+     * @since 4.13
+     */
+    @Override
+    protected final List<Description> orderItems(Collection<Description> descriptions) {
+        /*
+         * In practice, we will never get here--Sorters do their work in the
+         * compare() method--but the Liskov substitution principle demands that
+         * we obey the general contract of Orderable. Luckily, it's trivial to
+         * implement.
+         */
```

Comment:
Since this method is protected, I'm wondering whether we shouldn't just throw an `UnsupportedOperationException`. Thoughts?

Reply:
Since it's trivial to implement, and Ordering defines the method, I think we should implement it. A future subclass or Sorter could decide to call this method.